app/helpers/sheet/group/nav_left.rb

```diff
@@ -31,8 +31,31 @@ def render
 
       private
 
+      def sort_groups_by_name(groups)
+        children_of = {}
+        groups.each do |o|
+          children_of[o.parent_id] ||= []
+          children_of[o.parent_id] << o
+        end
+
+        children_of.each_value do |children|
+          children.sort_by!(&:sorting_name)
+        end
+
+        results = []
+        recombine_groups(results, children_of, groups.first.parent_id)
+        results
+      end
+
+      def recombine_groups(results, children_of, parent_id)
+        children_of[parent_id]&.each do |o|
+          results << o
+          recombine_groups(results, children_of, o.id)
+        end
+      end
+
       def groups
-        @groups ||= entry.groups_in_same_layer.without_deleted.order(:lft).to_a
+        @groups ||= sort_groups_by_name entry.groups_in_same_layer
```

Comment:
We could avoid sorting in cases where non of the groups have a translated `static_name`. But I'm not sure if the performance impact is big enough for us to care.